ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/PrivacyBlockProcessor.java

```diff
@@ -79,20 +80,23 @@ public Result processBlock(
       final MutableWorldState worldState,
       final BlockHeader blockHeader,
       final List<Transaction> transactions,
-      final List<BlockHeader> ommers) {
+      final List<BlockHeader> ommers,
+      final PrivateMetadataUpdater privateMetadataUpdater) {
+
+    if (privateMetadataUpdater != null) {
+      throw new IllegalArgumentException("PrivateMetadataUpdater passed in is not null.");
+    }
 
     maybeRehydrate(blockchain, blockHeader, transactions);
 
-    final PrivacyGroupHeadBlockMap privacyGroupHeadBlockMap =
-        new PrivacyGroupHeadBlockMap(
-            privateStateStorage
-                .getPrivacyGroupHeadBlockMap(blockHeader.getParentHash())
-                .orElse(PrivacyGroupHeadBlockMap.EMPTY));
-    privateStateStorage
-        .updater()
-        .putPrivacyGroupHeadBlockMap(blockHeader.getHash(), privacyGroupHeadBlockMap)
-        .commit();
-    return blockProcessor.processBlock(blockchain, worldState, blockHeader, transactions, ommers);
+    final PrivateMetadataUpdater metadataUpdater =
+        new PrivateMetadataUpdater(blockHeader, privateStateStorage);
+
+    final Result result =
+        blockProcessor.processBlock(
+            blockchain, worldState, blockHeader, transactions, ommers, metadataUpdater);
+    metadataUpdater.commit();
```

Comment:
At this point, has Besu already commit the changes in public world state?

Reply:
Yes, the public changes have been commited at the end of the wrapped block processor.

Reply:
... but the block has not been added to the chain yet. That is happening later.